src/addon/workbench/workbench_loader.js

```diff
@@ -0,0 +1,186 @@
+/**
+ * @fileoverview Workbench addon project loader
+ *
+ * @license Copyright 2018 The Coding with Chrome Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @author mdiehl@workbenchplatform.com (Matt Diehl)
+ */
+goog.provide('cwc.addon.WorkbenchLoader');
+
+goog.require('goog.net.XhrIo');
+goog.require('cwc.utils.Logger');
+
+
+/**
+ * @param {!cwc.utils.Helper} helper
+ * @param {!cwc.utils.Database} projectsDb
+ * @param {!cwc.utils.Database} imagesDb
+ * @constructor
+ * @struct
+ * @final
+ */
+cwc.addon.WorkbenchLoader = function(helper, projectsDb, imagesDb) {
+  /** @type {!string} */
+  this.name = 'WorkbenchLoader';
+
+  /** @private {!cwc.utils.Database} */
+  this.projectsDb_ = projectsDb;
+
+  /** @private {!cwc.utils.Database} */
+  this.imagesDb_ = imagesDb;
+
+  /** @type {!cwc.utils.Helper} */
+  this.helper = helper;
+
+  /** @private {!cwc.utils.Logger} */
+  this.log_ = new cwc.utils.Logger(this.name);
+
+  /** @private {!Array<Function>} */
+  this.loadCompleteListeners_ = [];
+
+  /** @private {string} */
+  this.projectsApiBase_ = 'https://staging.cwist.com/api/v1/activities/';
+
+  /** @private {string} */
+  this.projectsApiAll_ = `${this.projectsApiBase_}?content_channels=1`;
+};
+
+
+/**
+ * Loads projects from Workbench
+ */
+cwc.addon.WorkbenchLoader.prototype.loadProjects = function() {
+  if (!this.helper.checkFeature('online')) return;
+
+  let xhr = new goog.net.XhrIo();
+  goog.events.listen(xhr, goog.net.EventType.SUCCESS, (e) => {
```

Comment:
Would be great if you could add getUriAsJson to cwc.utils.Resources and use the function from here.